core/blockchain_test.go

```diff
@@ -471,11 +471,14 @@ func makeBlockChainWithDiff(genesis *types.Block, d []int, seed byte) []*types.B
 
 func chm(genesis *types.Block, db ethdb.Database) *BlockChain {
 	var eventMux event.TypeMux
-	bc := &BlockChain{chainDb: db, genesisBlock: genesis, eventMux: &eventMux, pow: FakePow{}, rand: rand.New(rand.NewSource(0))}
-	bc.headerCache, _ = lru.New(100)
+	bc := &BlockChain{chainDb: db, genesisBlock: genesis, eventMux: &eventMux, pow: FakePow{}}
+	bc.hc = &HeaderChain{chainDb: db, genesisHeader: genesis.Header(), rand: rand.New(rand.NewSource(0))}
+	bc.hc.getValidator = func() HeaderValidator { return bc.Validator() }
+	bc.hc.procInterrupt = bc.getProcInterrupt
```

Comment:
Please refactor the `bc` initialization to multiple lines and actually insert `hc` in there directly instead of an extra init step.

E.g.:

```go
bc := &BlockChain{
  chainDb: db,
  genesisBlock: genesis,
  ...
  hc: &HeaderChain{
    ...
  }
```